jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -165,26 +179,35 @@ public synchronized void execute() throws MojoExecutionException {
         // trigger initialization of the generator instance
         this.getGenerator();
 
+        boolean executedGenerator = false;
+
         if (this.classNames != null) {
             for (String className : this.classNames) {
                 this.getLog().info("Generating JSON Schema for <className>" + className + "</className>");
                 generateSchema(className, false);
+                executedGenerator = true;
             }
         }
 
         if (this.packageNames != null) {
             for (String packageName : this.packageNames) {
                 this.getLog().info("Generating JSON Schema for <packageName>" + packageName + "</packageName>");
                 generateSchema(packageName, true);
+                executedGenerator = true;
```

Comment:
**suggestion (blocking)**: This kind of variable re-assignment seems unnecessary.
Instead, check whether the `classNames´ or `packageNames` are neither `null` nor empty in one place.

Reply:
Changed that